### DISCUSSION TITLE
Run starlight on a subdomain while pages stay on the main domain?

### DISCUSSION THREAD

**Discussion by thansk** [withastro/starlight#1560] [Q&A] [answered]
With a similar structure to this https://github.com/Charca/fake-stripe/, I want to run my website (pages/) on example.org while the starlight pages (content/) go on blog.example.org

Is that possible and how?

**Comment by delucis** [comment]
I think it would depend entirely how you deploy the site.

In a traditional web server, it’s usually possible to point a subdomain pretty much anywhere, so you could have a build output like:

- docs/
  - index.html
  - something.html
- index.html
- login/index.html

And tell your server to point the `docs` subdomain at the `docs/` folder.

It might be trickier on services like Vercel and Netlify, I’m not sure how you’d do it there although I don’t want to rule it out.

**Reply by thansk** [reply]
The static export in that project puts the docs and the normal pages on the root
```
├── guides
│   └── example
│       └── index.html
├── login
│   └── index.html
```

I could nest it one more directory in the source inside `docs/`, so `docs/docs/` and use something like that in nginx

```nginx
server {
    listen 80;
    server_name docs.example.org;

    location / {
        root /usr/share/nginx/html/docs;
        index index.html;
    }
}

server {
    listen 80;
    server_name example.org;

    location / {
        root /usr/share/nginx/html;
        index index.html;
    }
}
```

but then in the deployed version, the docs.example.org looks for assets (_astro CSS files, etc.) in docs.example.org while they are on example.org or one directory up in the file system.

I used
```mjs
{
  build: {
    assetsPrefix: 'https://example.org'
  }
}
```
in the Astro config to force it, although that would break local builds but otherwise works fine, if I also allow CORS on the server.

Autogenerated sidebar links, however, don't work as they start pointing to the nested `/docs/reference/example/` instead of just `/reference/example/`

```mjs
{
  label: "Reference",
  autogenerate: {
    directory: "docs/reference",
  },
},
```

Is there a way to fix those?

**Reply by delucis** [reply, marked as answer]
Ahhhh, yeah the autogenerated links aren’t going to know how you’re deploying this. Obviously, one fix is to avoid the autogenerated groups and input sidebar items manually.

Otherwise, you could use a hack to mutate the data in [component overrides](https://starlight.astro.build/guides/overriding-components/):

```js
// src/overrides/removeBase.ts
import type { Props } from '@astrojs/starlight/props';

// Base path — replace “docs” with the base actually in use
const base = /^\/docs\//;
const stripBase = (path: string) => path.replace(base, '/');

/** Mutate props to remove base. */
export function removeBaseFromProps(props: Props) {
  props.sidebar = removeBaseFromSidebarGroup(props.sidebar);
  if (props.pagination.prev) {
    props.pagination.prev.href = stripBase(props.pagination.prev.href);
  }
  if (props.pagination.next) {
    props.pagination.next.href = stripBase(props.pagination.next.href);
  }
}

function removeBaseFromSidebarGroup(group: Props['sidebar']): Props['sidebar'] {
  return group.map((entry) => ({
    ...entry,
    ...(entry.type === 'group'
      ? { entries: removeBaseFromSidebarGroup(entry.entries) }
      : { href: stripBase(entry.href) }),
  }));
}
```

Then use that in overrides for the `Sidebar` and `Pagination` components:

```astro
---
// src/overrides/Sidebar.astro
import type { Props } from '@astrojs/starlight/props';
import Default from '@astrojs/starlight/components/Sidebar.astro';
import { removeBase } from './removeBase';

removeBase(Astro.props);
---

<Default {...Astro.props}><slot /></Default>
```
```astro
---
// src/overrides/Pagination.astro
import type { Props } from '@astrojs/starlight/props';
import Default from '@astrojs/starlight/components/Pagination.astro';
import { removeBase } from './removeBase';

removeBase(Astro.props);
---

<Default {...Astro.props}><slot /></Default>
```

And add these to your Starlight config in `astro.config.mjs`.

---

The other option to avoid all this would be to build your two sites separately. If you had two Astro projects — one for the subdomain, one for the primary domain — you might find it to work a bit more smoothly.

**Reply by thansk** [reply]
Very helpful, thank you. And easier than I thought it would be!

I will definitely consider two sites as these overrides are more likely to break with future major upgrades to Astro or to Starlight.
